lib/Dialect/TritonGPU/Transforms/Pipeliner/OuterLoopPipeline.cpp

```diff
@@ -55,8 +55,18 @@ createSchedule(scf::ForOp forOp, int numStages) {
 static void hoistAllocAndConst(scf::ForOp forOp) {
   SmallVector<Operation *> toHoist;
   for (Operation &op : forOp.getBody()->without_terminator()) {
-    if (isa<ttg::LocalAllocOp, arith::ConstantOp>(op))
+    if (auto allocOp = dyn_cast<ttg::LocalAllocOp>(op)) {
+      // We cannot hoist the allocOp if any of its operands
+      // is declared in the forOp's body.
+      if (llvm::any_of(allocOp->getOperands(),
+                         [&](Value operand) {
+                           return operand.getParentBlock() == forOp.getBody();
+                         }))
```

Comment:
actually can we skip any allocation that has an init value? I thought I did that but looks like I missed it.
Basically:
```
if (!allocOp.getInit())
  continue;
```

Reply:
Thanks for the quick review, Thomas! Would it be possible that the ```init``` argument of the ```allocOp``` comes from the parent scope? If that was the case, we might want something like the following?

```
      auto init = allocOp.getInit();
      if (!init || init.getParentBlock() != forOp.getBody())
        toHoist.push_back(&op);
```

Reply:
it would work but I don't think that's what we want to do. The idea here is to hoist the alloc created by the inner loop pipelining, we shouldn't try to move the other alloc ops.

Reply:
I see. Make sense. Updated the PR. Thanks!